fdbserver/ClusterRecovery.actor.cpp

```diff
@@ -461,8 +461,11 @@ ACTOR Future<Void> trackTlogRecovery(Reference<ClusterRecoveryData> self,
 	    self->configuration; // self-configuration can be changed by configurationMonitor so we need a copy
 	loop {
 		state DBCoreState newState;
-		self->logSystem->purgeOldRecoveredGenerations();
 		self->logSystem->toCoreState(newState);
```

Comment:
The purging needs to happen after "wait(self->cstate.write())" returns (down below) in this actor? If not, don't we still have the same problem that you described above?